src/google/protobuf/io/coded_stream.h

```diff
@@ -113,7 +113,9 @@
 #include <climits>
 #include <string>
 #include <utility>
-#ifdef _MSC_VER
+#if defined(__BYTE_ORDER__) && defined(__ORDER_LITTLE_ENDIAN__)
+  #define PROTOBUF_LITTLE_ENDIAN (__BYTE_ORDER__ == __ORDER_LITTLE_ENDIAN__)
```

Comment:
The callsite code actually checks for ```#if defined(PROTOBUF_LITTLE_ENDIAN)``` so should probably move this in to the line above.